packages/wallet/src/redux/indirect-funding/player-a/reducer.ts

```diff
@@ -18,19 +38,305 @@ export function playerAReducer(
+      );
+      const ledgerChannel = selectors.getOpenedChannelState(
+        newState,
+        indirectFundingState.ledgerId,
+      );
+      if (
+        hasConsensusBeenReached(
+          ledgerChannel.lastCommitment.commitment,
+          ledgerChannel.penultimateCommitment.commitment,
+        )
+      ) {
+        // Update app channel and send out any messages generated by the state change
+        updateChannelState(
+          state,
+          actions.internal.fundingConfirmed(indirectFundingState.channelId),
+        );
+      }
+      return newState;
+    default:
+      return state;
+  }
+};
+
+const waitForPostFundSetup1 = (
+  state: walletStates.Initialized,
+  action: actions.indirectFunding.Action,
+): walletStates.Initialized => {
+  switch (action.type) {
+    case actions.COMMITMENT_RECEIVED:
+      const newState = { ...state };
+      const indirectFundingState = selectors.getIndirectFundingState(
+        state,
+      ) as states.WaitForPostFundSetup1;
+
+      // Update  ledger channel
+      updateChannelState(
+        newState,
+        channelActions.opponentCommitmentReceived(action.commitment, action.signature),
+      );
+
+      const ledgerChannel = selectors.getOpenedChannelState(
+        newState,
+        indirectFundingState.ledgerId,
+      );
+      if (ledgerChannel.type !== channelStates.WAIT_FOR_UPDATE) {
+        return newState;
+      } else {
+        // If the ledger channel is funded we can create the allocation request
+        const appChannel = selectors.getOpenedChannelState(
+          newState,
+          indirectFundingState.channelId,
+        );
+        const proposedAllocation = [
+          calculateChannelTotal(appChannel.lastCommitment.commitment.allocation),
+        ];
+        const proposedDestination = [appChannel.channelId];
+
+        const { updateCommitment, commitmentSignature } = composeLedgerUpdateCommitment(
+          ledgerChannel.lastCommitment.commitment,
+          ledgerChannel.ourIndex,
+          proposedAllocation,
+          proposedDestination,
+          ledgerChannel.privateKey,
+        );
+
+        // Update our ledger channel with the latest commitment
+        updateChannelState(newState, channelActions.ownCommitmentReceived(updateCommitment));
+        // Send out the commitment to the opponent
+        newState.outboxState.messageOutbox = [
+          createCommitmentMessageRelay(
+            ledgerChannel.participants[PlayerIndex.B],
+            indirectFundingState.ledgerId,
+            updateCommitment,
+            commitmentSignature,
+          ),
+        ];
+        return newState;
+      }
+
+    default:
+      return state;
+  }
+};
+const waitForDirectFunding = (
+  state: walletStates.Initialized,
+  action: actions.indirectFunding.Action,
+) => {
+  const indirectFundingState = selectors.getIndirectFundingState(
+    state,
+  ) as states.WaitForDirectFunding;
+  switch (action.type) {
```

Comment:
This should probably be
```
if (!isFundingAction(action)) {
  return state 
}
/// other logic
```

The idea here is that
- the indirect funding coordinator relays any funding related action to `updateFundingState`
- it then checks if the channel is funded
  - if yes, do some other stuff

Reply:
Updated